static/js/common/upload-addon.js

```diff
@@ -431,6 +431,20 @@
                       $("<p>").text(gettext("Your submission will be automatically signed.")).appendTo(upload_results);
                     }
 
+                    if (results.validation.is_upgrade_to_webextension) {
+                        var warning_box = $('<div>').attr('class', 'important-warning').appendTo(upload_results);
```

Comment:
If I read the code above correctly this looks like you're appending to the DOM and then modifying it. It would be better to build up everything and append it in one go instead.